backend/uclapi/dashboard/tests.py

```diff
@@ -442,3 +446,112 @@ def test_refresh_verification_secret_POST_success(
         self.assertEqual(response.status_code, 200)
         self.assertTrue(content["ok"])
         self.assertTrue("new_secret" in content.keys())
+
+
+def post_request_only(self, url, view):
+    request = self.factory.get(
+        url,
+        {
+        }
+    )
+
+    response = view(request)
+    content = json.loads(response.content.decode())
+    self.assertEqual(response.status_code, 400)
+    self.assertEqual(
+        content["error"],
+        "Request is not of method POST"
+    )
+
+
+def empty_get_request_only(self, url, view, error):
+    request = self.factory.post(
+        url,
+        {
+        }
+    )
+
+    response = view(request)
+    content = json.loads(response.content.decode())
+    self.assertEqual(response.status_code, 400)
+    self.assertEqual(
+        content["message"],
+        error
+    )
+
+
+class api_applicationsTestCase(TestCase):
+    def setUp(self):
+        self.factory = APIRequestFactory()
+        self.functions = {
+            '/api/create/': (create_app, 0), '/api/rename/': (rename_app, 1),
+            '/api/regen/': (regenerate_app_token, 2),
+            '/api/delete/': (delete_app, 2),
+            '/api/setcallbackurl/': (set_callback_url, 2),
+            '/api/updatescopes/': (update_scopes, 2)
+        }
+
+    def test_get_user_returns_correct_user(self):
+        user_ = User.objects.create(
+            email="test@ucl.ac.uk",
+            cn="test",
+            given_name="Test Test"
+        )
+        self.assertEqual(get_user_by_id(user_.id), user_)
+
+    def test_get_request_rejected(self):
+        for url in self.functions:
+            post_request_only(self, url, self.functions[url][0])
+
+    def test_missing_parameters(self):
+        errors = (
+            "Request does not have name or user.",
+            "Request does not have app_id/new_name",
+            "Request does not have an app_id."
+        )
+        for url in self.functions:
+            empty_get_request_only(
+                self,
+                url,
+                self.functions[url][0],
+                errors[self.functions[url][1]]
+            )
+
+    def test_missing_parameters2(self):
+        request = self.factory.post(
+            '/api/create/',
+            {
+            }
+        )
+
+        response = create_app(request)
+        content = json.loads(response.content.decode())
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(
+            content["message"],
+            "Request does not have name or user."
```

Comment:
This is a gross error message. Might be worth changing it in the view too?

Reply:
(or perhaps not if this isn't even shown to the user anyway, which is quite possible if it gets sent to the Dashboard)

Reply:
Yeh, although theoretically someone could use the dashboard via this api without the frontend.

Reply:
That is true. I am actually no opposed to this if somebody *really* wanted to. I'm happy to replace the Dashboard at this point because it's quite dated and doesn't really support the developers much.